packages/postgres-database/src/simcore_postgres_database/utils_aiopg_orm.py

```diff
@@ -156,35 +156,80 @@ async def fetch(
         if rowid:
             # overrides pinned row
             query = query.where(self._primary_key == rowid)
-        elif self.is_default_set():
-            assert self._unique_match is not None  # nosec
-            query = query.where(self._unique_match)
+        elif self.is_filter_set():
+            assert self._where_clause is not None  # nosec
+            query = query.where(self._where_clause)
 
         result: ResultProxy = await self._conn.execute(query)
         row: Optional[RowProxy] = await result.first()
         return row
 
-    async def fetchall(
+    async def fetch_all(
         self,
         returning_cols: Union[str, List[str]] = ALL_COLUMNS,
     ) -> List[RowProxy]:
 
         query = self._compose_select_query(returning_cols)
+        if self.is_filter_set():
+            assert self._where_clause is not None  # nosec
+            query = query.where(self._where_clause)
 
         result: ResultProxy = await self._conn.execute(query)
         rows: List[RowProxy] = await result.fetchall()
         return rows
 
+    async def fetch_page(
+        self,
+        returning_cols: Union[str, List[str]] = ALL_COLUMNS,
+        *,
+        offset: int,
+        limit: Optional[int] = None,
+        order=None,
+    ) -> Tuple[List[RowProxy], int]:
+        """Support for paginated fetchall
+
+        IMPORTANT: pages are sorted by primary-key
```

Comment:
in the case of paging it is not an option ... it is about consistency, right? you need an order.

Reply:
Since the data will always be listed in the same way if there are no new elements added to the set, I think the default could be omitted. In theory it should use the primary for indexing. Omitting the sort will skip some steps form the internal query optimiser.

Reply:
based on the [tests i have performed](https://github.com/ITISFoundation/osparc-simcore/blob/46be7bcacb2af46bd2c1436dd37ee3bf5194e712/packages/postgres-database/tests/test_utils_aiopg_orm.py), my guess is that the order is the insertion order. Normally the primary key is an auto-incremented index and therefore has no effect, but if the primary key is e.g. a unique name, then the order is NOT folloing the primary key.